### DISCUSSION TITLE
Snap marker to (for example) highest value nearby

### DISCUSSION THREAD

**Discussion by rhult** [patrykandpatrick/vico#620] [Ideas]
I am trying to implement a marker that snaps to the highest y value nearby the clicked position. I am looking at the getClosestMarkerEntryModel function, which looks like where the current behavior is defined, but I can't find any way to customize it. Does anyone have any idea about this?

**Comment by patrickmichalik** [comment]
Hello! Just to confirm, which of the following is the case?

1. Multiple points are currently highlighted. You’d like only the topmost one of these to be marked.
2. Currently, the point that you’d like to be highlighted is not marked at all.

**Reply by rhult** [reply]
I find it a bit difficult to explain, but let me try: I'm displaying a chart that looks similar to this [example](
https://www.researchgate.net/publication/328358018/figure/fig2/AS:683070361444367@1539868073914/FFT-graphs-for-motor-Y-axis-without-damper-left-and-with-damper-right.png). I have a marker with the value, that is shown when the user taps or drags across the graph. The interesting values for the user are the various peaks in the chart, but it is very difficult or impossible to hit the actual peaks. So what I'm trying to do is to "snap" to the highest value near the click (say within the surrounding 5 values).

**Reply by patrickmichalik** [reply]
Got it, @rhult. I’m afraid this isn’t possible yet, but we’ll look into adding an API that facilitates implementing such behavior. I’m moving the discussion to the “Ideas” category.

**Reply by rhult** [reply]
Great, thanks!

**Reply by Gowsky** [reply]
Hello, @rhult. While the dedicated API mentioned by @patrickmichalik hasn't been added yet, this is possible since Vico `2.0.0-alpha.23` via subclassing. You can override `CartesianChart.drawMarker` to implement the kind of snapping you've described.
